### ISSUE TITLE
Feature - Split content, use of 'data-sources'

### ISSUE THREAD

**Issue by janvanhelvoort** [umco/umbraco-stacked-content#53]
Hello,

in case of 

> Please ask first before embarking on any significant pull request (e.g. implementing features, refactoring code, porting to a different language), otherwise you risk spending a lot of time working on something that the project's developers might not want to merge into the project.

A couple of years ago, I have worked with sitecore. One key feature there was use of datasources. For future sites we will use this package instead of the grid for the page layout / content.

I have an idea how we could add this function to stacked content, if it fits in the scope of this project.

I would like to add a option / setting to a stacked content property where you could enable datasource, something like: `Select content (node) enabled` which is a checkbox.

If you don't check this box, nothing will change.

If you check the box, we could render a content picker above the fields, which is restricted to the allowed content type of the stack.
If a content node is picked, the editor fields are filled with the content of the node, but the fields would be disabled, to show there are from the source. 
When rendering, if content is picked, serve the content from the node, otherwise the content created at the stacked content node. Maybe this is possible inside the property value converter.

other possibilities:
- A/B testing with multiple content pickers?
- Behavioral targeting with rules, 
  - If it is weekend or evening, show other content
  - If the user is logged in

If it fit in the scope of this package, I will give it a try.

Thanks in advance,
Jan van Helvoort

**Comment by dawoe** [comment]
Hi @janvanhelvoort 

You can already do this without changing the core package. Create a datasources folder doctype...and for allowed children you can set the same doctypes used by stacked content.

Then create a doctype for stacked content with a content picker or MNTP or whatever... and let the editor pick a item from the datasource folder

This will allow for reuse of stacked content blocks accross pages.

Because you probably already have partial views/ controller actions it would be easy to do the correct rendering based on the doctype of the selected item.

At least that's how I solved it for doctype grid editor

Dave

**Comment by janvanhelvoort** [comment]
Hi @dawoe 

Interesting approach, I will give it a look.

But if you will give them the opportunity to choose between both options at the same time? If you only need to create the content item once, you will need to create it first at the datasource folder?

It was an idea what I get when i was walking, at this moment I did not work yet with stacked content. I love nested content and used it a lot, we will replace the grid at next projects.

if it will not fit in the scope of this package, we need to rethink a work around. At this moment I have some spare time to create something.

Jan

PS: Rub it in 👍

**Comment by dawoe** [comment]
Hi Jan,

Maybe something like this package could help you : https://our.umbraco.com/packages/backoffice-extensions/doctype-grid-editor-reusable-content-extension/

This is for doctype grid editor, but the same principle could be applied to stacked content.

But haven't tried that package yet

Dave

**Comment by janvanhelvoort** [comment]
Thank you Dave, 

The package, doctype-grid-editor-reusable-content-extension, doesn't work with newer versions, so i can't even tried it.

I will wait for an answer from the package maintainers.

Jan

**Comment by leekelleher** [comment]
@janvanhelvoort Thanks for the suggestion!

We're trying to understand the wider use-case for such a feature, it probably needs some further discussion and exploration, (before adopting it).

We did add a feature to InnerContent (which is available in StackedContent v2.0) for using Content Templates (aka Blueprints).  Could that be of use to you?

**Comment by janvanhelvoort** [comment]
Hi @leekelleher,

Yes, the use of Content Templates could work, but the editor, as far I know, can't edit or change the template without access to the settings tab?

The use case for the content picker would something like, on multiple pages you have a contact section, with e-mail and phone-numbers, if you want add of change this content regularly, you would need to go to evey page and change (or copy and paste) the changed values.

If you could link a node to a stack, you only need to change the content of thad node and publish it. Every stack with the linked content, would show the correct content.

**Comment by mattbrailsford** [comment]
I think Daves suggestion is what you want and it's an approach I often use so in my stacked content pre value I'll have a list of doctypes I can use for inline use, but I'll also have a doctype called something like "Linked Content" and that doc type has a picker on it to pick content from a shared repo. You are correct in thinking when using this approach that you would need to create the content in the repo first however before you can create it.

**Comment by leekelleher** [comment]
Closing this ticket off. Thank you for the suggestion @janvanhelvoort, we do appreciate it! ❤️ 

I can definitely see the use-cases for this feature ... _fun fact! it was a feature that we added to our unreleased [Mortar editor](https://github.com/leekelleher/umbraco-mortar), of which we'd ported features over to DTGE, I digress!_  However, for Stacked Content, we'd like to keep the scope as is, with doctypes to configure blocks.
